molten-cache/src/main/java/com/hotels/molten/cache/metrics/CaffeineCacheStatsInstrumenter.java

```diff
@@ -109,16 +132,13 @@ public void recordEviction(int weight, RemovalCause cause) {
     @Nonnull
     @Override
     public CacheStats snapshot() {
-        return new CacheStats(Double.valueOf(hitCount.count()).longValue(), Double.valueOf(missCount.count()).longValue(), loadSuccess.count(), loadFailure.count(),
+        return CacheStats.of(Double.valueOf(hitCount.count()).longValue(), Double.valueOf(missCount.count()).longValue(), loadSuccess.count(), loadFailure.count(),
             totalLoadTime.longValue(), Double.valueOf(evictionCount.count()).longValue(), totalEvictedWeight.longValue());
     }
 
-    /**
-     * Registers the related {@link Cache} to report its size as well.
-     *
-     * @param cache the cache to register size gauge for
-     */
-    public void registerCache(Cache cache) {
-        metricId.extendWith("size").toGauge(cache, Cache::estimatedSize).register(meterRegistry);
+    private void warnIfUnregistered() {
+        if (!registered) {
+            LOG.warn("Unregistered cache found with metricsQualifier={}. Estimated size metrics won't be available!", metricId.getHierarchicalName());
```

Comment:
This might spam the log. I would introduce a `unregisteredCacheReported` flag and log it only once.

Reply:
Changed it in 7ac0cc65 to log only once, but I think it should be switchable, because one single warn per cache should make no harm when it's not used properly.